iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/Properties.java

```diff
@@ -0,0 +1,171 @@
+        }
+        return Optional.ofNullable(returnValue);
+    }
+
+    public <P> Optional<P> getProperty(String key, P defaultValue) {
+        Optional<P> val = getProperty(key);
+        return Optional.of(val.orElse(defaultValue));
+    }
+
+    public static boolean isSameType(Properties.Property oldValue, Properties.Property newValue) {
+        return newValue.getType() == oldValue.getType();
+    }
+
+    public Set<String> keySet() {
+        return propertyList.keySet();
+    }
+
+    public void mergeProperty(DanglingLine dl1, DanglingLine dl2, String name, Type type) {
+        Optional dl1Property = dl1.getProperty(name);
+        Optional dl2Property = dl2.getProperty(name);
+
+        if (Objects.equals(dl1Property, dl2Property)) {
+            if (dl1Property.isPresent()) {
+                put(name, new Property(type, dl1Property.get()));
+            }
+        } else if (dl1Property.isPresent() && (!dl2Property.isPresent() ||
+            dl2.getPropertyType(name) == Type.STRING && ((String) dl2Property.get()).isEmpty())) {
+            LOGGER.warn(INCONSISTENCY_WARN_EMPTY_SIDE_2, name, dl1Property.get());
+            put(name, new Property(type, dl1Property.get()));
+        } else if (dl2Property.isPresent() && (!dl1Property.isPresent() ||
+            dl2.getPropertyType(name) == Type.STRING && ((String) dl1Property.get()).isEmpty())) {
+            LOGGER.warn(INCONSISTENCY_WARN_EMPTY_SIDE_1, name, dl2Property.get());
+            put(name, new Property(type, dl2Property.get()));
+        } else {
+            LOGGER.error(INCONSISTENCY_ERROR_BOTH_SIDES, name, dl1Property, dl2Property);
+        }
+    }
+
+    private void setDanglingLineProperty(Map<String, Property> properties, DanglingLine dl, String name) {
+        Type type = dl.getPropertyType(name);
+        dl.getProperty(name).ifPresent(o -> properties.put(name, new Property(type, o)));
+    }
+
+    public void mergeProperties(DanglingLine dl1, DanglingLine dl2) {
+        Set<String> dl1Properties = dl1.getPropertyNames();
+        Set<String> dl2Properties = dl2.getPropertyNames();
+        Set<String> commonProperties = Sets.intersection(dl1Properties, dl2Properties);
+        Sets.difference(dl1Properties, commonProperties).forEach(prop -> setDanglingLineProperty(propertyList, dl1, prop));
+        Sets.difference(dl2Properties, commonProperties).forEach(prop -> setDanglingLineProperty(propertyList, dl2, prop));
+        commonProperties.forEach(prop -> {
+            if (dl1.getPropertyType(prop).equals(dl2.getPropertyType(prop))) {
+                mergeProperty(dl1, dl2, prop, dl1.getPropertyType(prop));
+            } else {
+                LOGGER.error("Inconsistencies of property type for '{}' between both sides of merged line. '{}' on side 1 and '{}' on side 2. Removing the property of merged line",
+                    prop, dl1.getPropertyType(prop), dl2.getPropertyType(prop));
+            }
+        });
+    }
```

Comment:
I'm not sure these methods should be there: it's too specific. Why do not move it to the DanglingLine class.